Facebooky/ApplyShortcutsLink.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Specialized;
+
+namespace Facebooky
+{
+	class ApplyShortcutsLink : IPostChainLink
+	{
+		public IPostChainLink NextLink { get; set; }
+		public bool Enabled { get; set; }
+
+		private readonly StringDictionary r_ShortcutsToReplace;
+
+		public ApplyShortcutsLink(StringDictionary i_ShortcutsToReplace)
+		{
+			this.r_ShortcutsToReplace = i_ShortcutsToReplace;
+		}
+
+		public void HandlePost(PostInfo i_Post)
+		{
+			if (Enabled)
+			{
+				if (r_ShortcutsToReplace.Count > 0)
+				{
+					foreach (string key in r_ShortcutsToReplace.Keys)
+					{
+
+						i_Post.StatusText = i_Post.StatusText.Replace(key, r_ShortcutsToReplace[key]);
+					}
+				}
+			}
+			if (NextLink != null)
```

Comment:
This condition is being reused in every link.
Also You force the command to be a chain link.

What you should do is create a ChainCommand that wraps any ICommand turning into a ChainLink.
Have it's execute be:

```c#
void Execute()
{
    this.Command.Execute();
    if (this.NextCommand != null)
    {
        this.NexCommand.Exectute();
    }
}
```
That way you reuse the chain logic and enable having the commands as chained or not.

![UML](http://www.plantuml.com/plantuml/svg/VL0x3i8m3DrzYeo0r0jqAYeCB5quWPKO6ZMVAd5f4UBkH4KDKgNYoVPxzdjs5Ha33zO0i6P3eZOO8qLHV3DnqdWFQ06qOmetb2GkjRSMdH8l42dE4yc1QRSl7ruhU0F8UR9kKJjI2rdRtf0bnt6zOAdA3CpjXYRUWB9a0XRHbToXK7vqnMykkoBPDSIZ3zr_Q_cGOz-RvxNrWPDet6PNv5Jwv0S0)